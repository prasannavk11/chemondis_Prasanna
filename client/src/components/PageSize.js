import React from "react";

const PageSize = (props) => {
  const { sizes, count, onChangePageSize, disabled } = props;
  return (
    <nav aria-label="Page navigation example">
      <ul className="pagination justify-content-end">
        {sizes &&
          sizes.map((size, index) => {
            return (
              <li
                key={index}
                className={disabled ? "page-item disabled" : "page-item"}
                onClick={onChangePageSize}
              >
                <div
                  className={
                    count === size
                      ? "page-link cursor-pointer active"
                      : "page-link cursor-pointer"
                  }
                >
                  {size}
                </div>
              </li>
            );
          })}
      </ul>
    </nav>
  );
};

export default PageSize;
