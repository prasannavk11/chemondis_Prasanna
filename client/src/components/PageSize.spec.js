import { render } from "@testing-library/react";
import PageSize from "./PageSize";

afterEach(() => {
  jest.clearAllMocks();
  jest.resetAllMocks();
});

describe("PageSize", () => {
  it("should render successfully", () => {
    const { baseElement } = render(
      <PageSize
        sizes={[20, 30, 50]}
        count={10}
        onChangePageSize={() => {}}
        disabled={false}
      />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
