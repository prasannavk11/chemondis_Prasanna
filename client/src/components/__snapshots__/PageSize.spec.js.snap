// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSize should render successfully 1`] = `
<body>
  <div>
    <nav
      aria-label="Page navigation example"
    >
      <ul
        class="pagination justify-content-end"
      >
        <li
          class="page-item"
        >
          <div
            class="page-link cursor-pointer"
          >
            20
          </div>
        </li>
        <li
          class="page-item"
        >
          <div
            class="page-link cursor-pointer"
          >
            30
          </div>
        </li>
        <li
          class="page-item"
        >
          <div
            class="page-link cursor-pointer"
          >
            50
          </div>
        </li>
      </ul>
    </nav>
  </div>
</body>
`;
